[DOCS-10034] Update Browser SDK beforeSend callback function details (#27763)
DataDog/documentation: content/en/logs/log_collection/javascript.md

If your Browser logs contain sensitive information that needs redacting, configure the Browser SDK to scrub sensitive sequences by using the `beforeSend` callback when you initialize the Browser Log Collector.

The `beforeSend` callback function takes two arguments: the `log` event and `context`. The function gives you access to each log collected by the Browser SDK before it is sent to Datadog, and lets you update any property.

```javascript
function beforeSend(log,context)
```

> The potential `context` values are:

Reviewer: What does this function do with the `context` argument—why would a user set it, and what is the effect of setting it? 

I'm guessing it's for if a user wanted to only apply the `beforeSend` function to a subset of logs that match a specific context. When they set `context`, the `beforeSend` function applies only to logs that match `context`. Whatever it is, we should explicitly state it.

Author: Update in commit.

> The `beforeSend` callback function that you can provide and can be invoked with two arguments: the `log` event and `context`. The function gives you access to each log collected by the Browser SDK before it is sent to Datadog, and lets you use the context to adjust any log properties. The context contains additional information related to the event, but not necessarily included in the event. You can typically use this information to [enrich][18] your event or [discard][19] them.

Reviewer: ```suggestion
The `beforeSend` callback function can be invoked with two arguments: the `log` event and `context`. This function gives you access to each log collected by the Browser SDK before it is sent to Datadog, and lets you use the context to adjust any log properties. The context contains additional information related to the event, but not necessarily included in the event. You can typically use this information to [enrich][18] your event or [discard][19] it.
```

> | Value | Use Case |
> |-------|---------|
> | `isAborted` | For network log events. |
> | `handlingStack` | For network, console, and logger events - this value shows where the log event was handled, which is useful for micro-frontends. |

Reviewer: What does `isAborted` indicate? For `handlingStack`, what does "where" mean—a location? Also, we should probably specify data type. For example (and these are just my guesses at filling in missing information):

```suggestion
| Value | Data Type | Meaning | Use Case |
|-------|---------|
| `isAborted` | Boolean | Whether the log event is aborted. | For network log events. |
| `handlingStack` | String | Where the log event was handled, which is useful for micro-frontends. For example... <insert example values> | For network, console, and logger events. |
```

It doesn't need to be in this exact format

Reviewer 2: A agree this might need some clarification!

For `isAborted`, I would say "For network error logs, whether the failing request was aborted by the application."

For `handlingStack`, I would say "Where the log event was handled. Can be used to identify from which micro-frontend the log was sent." and maybe add a link to the [RUM doc](https://docs.datadoghq.com/real_user_monitoring/browser/advanced_configuration/?tab=npm#micro-frontend) or have a similar section for Logs.

> | `isAborted` | Boolean | For network log events, this property tells you whether the failing request was aborted by the application, in which case you might not want to send this event because it might be intentionally aborted. |

Author: ```suggestion
| `isAborted` | Boolean | For network log events, this property tells you whether the failing request was aborted by the application, in which case you might not want to send this event because it may be intentionally aborted. |
```

> | `isAborted` | Boolean | For network log events, this property tells you whether the failing request was aborted by the application, in which case you might not want to send this event because it might be intentionally. |

Reviewer: intentionally...? looks like part of the sentence is missing

Author: oops! addressed.
